### DISCUSSION TITLE
E3V2-BLTouch-5x5-HS-v4.2.7-v2.0.1.bin ... help

### DISCUSSION THREAD

**Discussion by john3d1977** [Jyers/Marlin#1787] [Q&A]
im a rookie i know... i have this firmware and  love it but i have a micro swiss hot end and want to hit at least 280c with it and it wont go above 260c  .... im not a programmer ,, i know my way around to a extent but get lost in programming.... whats my options

**Comment by BlueWings172** [comment]
@john3d1977 

You need to change this in the firmware code and then compile it. Here is what you need to do: 


1. Watch this video on how to compile firmware. It's very easy and will serve you for years. https://www.youtube.com/watch?v=so0FfFVEmiw&ab
2. Download version 2.02h of the firmware that @tititopher68-dev updated in this post #1694
3. Note in the in Configuration Files folder, there isn't 5x5 HS but there is 3x3 HS. So you use the 3x3 as a basis and start by copying the 2 files inside the BLTouch-3x3-HighSpeed folder to the marlin folder, replacing the 2 files already there.
4. We need to change the 3x3 configuration to 5x5. So as the video shows in Visual Studio Code, in Configuration.h, you need to change this line (1607):

`#define GRID_MAX_POINTS_X 3
`
to 

`#define GRID_MAX_POINTS_X 5
`
you can use Ctrl+F to locate the line quickly.

5. Now, we need to change the code to enable you to print at 280c. But note that Marlin does not allow to print at the temperature you set in the firmware. Instead, it allows to print at the value you insert - 15c as safety measure. So if you want print at 280, you have to set the value in the firmware at 280 + 15 = 295. So change line 560 as follows:

from:

`#define HEATER_0_MAXTEMP 275
`
to

`#define HEATER_0_MAXTEMP 295
`
6. Compile as shown in the video and flash it.
7. Take necessary precautions to ensure that it is safe to print at that temperature.

**Reply by john3d1977** [reply]
yes tried that but get a yellow code unknown oserror exeption code 22   c/windows  cmd/exe invalid argument.... and a red code pio build stmbuild error..127.. and the build failed...I retried and redownloaded and know I did it correct .. what am i missing and what do those error codes mean

**Reply by BlueWings172** [reply]
not sure. try compiling without making any changes to the code and see if you get errors. This sounds like an issue with your setup rather than the firmware.

**Reply by john3d1977** [reply]
Yes I tried that to and got the same issue...
